nvtabular/io/dataset.py

```diff
@@ -1051,6 +1052,22 @@ def regenerate_dataset(
         else:
             return result
 
+    def infer_schema(self, n=1):
+        """Create a schema containing the column names and inferred dtypes of the Dataset
+
+        Args:
+            n (int, optional): Number of rows to sample to infer the dtypes. Defaults to 1.
+        """
+        sampled_dtypes = self.sample_dtypes(n)
```

Comment:
:+1: 
Long term I don't think we'll need to sample the dtypes (since we should have correct dtype information on the Dataset ddf after workflow.transform calls)   - but I think this is a the right thing to do for now.

Reply:
We need to know the column names and dtypes before `Workflow.fit/transform` in order to compute which columns to read from the ddf though. 🙁

Reply:
The 'sample_dtypes' function was created originally because the dtype property on the ddf wasn't correct after going through workflow.transform (since we only propagated column names through the workflow graph, not full schema information). 

For getting the column names and dtype information for the input dataset, we don't need to sample at all - and could just use the dtypes on the ddf itself (like as done here as a fallback https://github.com/NVIDIA/NVTabular/blob/e9c0795ab68d4d42b13d3a73e888c73b1483fb73/nvtabular/io/dataset.py#L1067 ) .  The problem is that we can't currently rely on this for the output of the workflow graph, since the dtypes are getting mangled in sections of code like https://github.com/NVIDIA/NVTabular/blob/e9c0795ab68d4d42b13d3a73e888c73b1483fb73/nvtabular/workflow/workflow.py#L401-L407

Reply:
Yeah, seems like getting the dtypes of the input dataset is the main use for `infer_schema`, but I suppose it's possible you'd want to run multiple workflow transforms in series, in which case reading the dtypes directly from the ddf wouldn't work so well. 🤔